assets/shaders/defaultshaders.ini

```diff
@@ -144,4 +144,8 @@ SettingDefaultValue4=1.0
 SettingMaxValue4=2.0
 SettingMinValue4=0.1
 SettingStep4=0.05
-
+[Tex4xBRZ]
+Type=Texture
+Name=4xBRZ
+Author=Hyllian
+Compute=tex_4xbrz.csh
```

Comment:
Since the scaling shader is hardcoded for a specific scale factor (4x in this case), I think that should be specified here somehow, and checked / enforced in the VK texture cache code.

Reply:
It's kinda like having a different "output resolution", but I suppose at least it just not be forced for all shader types.

I guess it would just ignore your configured scale then?

-[Unknown]

Reply:
Yes, something like that. What does this do right now if the texture scale isn't set to 4x?

Reply:
It just runs the shader for each pixel, so it works fine.

In other words, in basically the same way the "5xBR Upscaler" postshader works when you don't use 5x render resolution (which also does not force anything.)

-[Unknown]

Reply:
Ah right, so since it uses the mix function at the end to choose the pixel, we end up with a bilinearly downsampled version of the 4x upscale. I supposed that's alright.

Reply:
Citra uses xBRZ freescale which is resolution independent, but I'm not sure how much work would be required to get it working without linear sampling.

Reply:
I'm starting to think that the overhead of one more copy and creating a temporary Image isn't actually all that high. In that case we can get free bilinear. Might do that soon.

Reply:
I think that's probably a wise idea 👍